apps/wallet/src/ui/app/pages/accounts/ProtectAccountPage.tsx

```diff
@@ -46,12 +49,15 @@ export function ProtectAccountPage() {
 			typeof hasPasswordAccounts !== 'undefined' &&
 			!(createMutation.isSuccess || createMutation.isLoading)
 		) {
-			setShowVerifyPasswordView(hasPasswordAccounts);
+			setShowVerifyPasswordView(hasPasswordAccounts && !isResetting);
 		}
-	}, [hasPasswordAccounts, createMutation.isSuccess, createMutation.isLoading]);
+	}, [hasPasswordAccounts, createMutation.isSuccess, createMutation.isLoading, isResetting]);
 	const createAccountCallback = useCallback(
 		async (password: string, type: CreateType) => {
 			try {
+				if (isResetting && type.includes('mnemonic')) {
+					await deleteMutation.mutateAsync({ type: 'mnemonic' });
```

Comment:
Deleting everything else before creating the new account probably is the faster approach for the use case here but I was thinking to do something that will allow users just update the password for the selected account and then delete the rest of the accounts because we need to keep the same password.

But doing it this way we could do things like, verify that, in case of mnemonic, it is the same as the existing one and maybe keep any derived accounts from this mnemonic there